pkg/controller/storagecapabilities.go

```diff
@@ -53,13 +56,49 @@ var CapabilitiesByProvisionerKey = map[string][]StorageCapabilities{
 	"csi.trident.netapp.io/ontap-san": {{AccessMode: v1.ReadWriteOnce, VolumeMode: v1.PersistentVolumeBlock}},
 }
 
-// Get finds and returns a predefined StorageCapabilities for a given StorageClass
-func Get(sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
+// GetStorageCapabilities finds and returns a predefined StorageCapabilities for a given StorageClass
+func GetStorageCapabilities(cl client.Client, sc *storagev1.StorageClass) ([]StorageCapabilities, bool) {
 	provisionerKey := storageProvisionerKey(sc)
+	if provisionerKey == "kubernetes.io/no-provisioner" {
+		return capabilitiesForNoProvisioner(cl, sc)
```

Comment:
So here we do a full scan of all PVs (filtering by StorageClassName), and we do it each time when GetStorageCapabilities is called, which is on every profile reconcile loop. Can we cache some data? would it be safe to cache it?

Reply:
Given that there's a field called "uncachedClient" and I am using the client I assumed it's cached.
@mhenriks - do you know if this is problematic to query on a larger cluster?

Reply:
Right, so maybe that is not problematic at all.